waku/v2/node/wakunode2.nim

```diff
@@ -130,25 +130,32 @@ template tcpEndPoint(address, port): auto =
 template addWsFlag() =
   MultiAddress.init("/ws").tryGet()
 
+template addWssFlag() =
+  MultiAddress.init("/wss").tryGet()
+
 
 proc new*(T: type WakuNode, nodeKey: crypto.PrivateKey,
     bindIp: ValidIpAddress, bindPort: Port,
     extIp = none[ValidIpAddress](), extPort = none[Port](),
     peerStorage: PeerStorage = nil,
     maxConnections = builders.MaxConnections,
     wsBindPort: Port = (Port)8000,
-    wsEnabled: bool = false): T 
-    {.raises: [Defect, LPError].} =
+    wsEnabled: bool = false,
+    wssEnabled: bool = false,
+    secureKey: string = "",
+    secureCert: string = ""): T 
+    {.raises: [Defect, LPError, IOError,TLSStreamProtocolError].} =
   ## Creates a Waku Node.
   ##
   ## Status: Implemented.
   ##
   let
     rng = crypto.newRng()
     hostAddress = tcpEndPoint(bindIp, bindPort)
-    wsHostAddress = tcpEndPoint(bindIp, wsbindPort) & addWsFlag
+    wsHostAddress = if wssEnabled : tcpEndPoint(bindIp, wsbindPort) & addWssFlag
+                    else : tcpEndPoint(bindIp, wsbindPort) & addWsFlag
     announcedAddresses = if extIp.isNone() or extPort.isNone(): @[]
-                        elif wsEnabled == false: @[tcpEndPoint(extIp.get(), extPort.get())]
+                        elif wsEnabled == false and wssEnabled == false: @[tcpEndPoint(extIp.get(), extPort.get())]
                         else : @[tcpEndPoint(extIp.get(), extPort.get()),
                         tcpEndPoint(extIp.get(), wsBindPort) & addWsFlag]
```

Comment:
Indentation. Bring this line in line with `tcpEndPoint` above so it's clearer to read as the next item in the `seq`.